### PR TITLE
Support domainname

### PR DESCRIPTION
**Pull request by higuruchi**
Add: #1213 

- [x] Implement
- [x] Add a test for it

- config.json
  ```diff_config.json
  {
    "ociVersion": "1.0.2-dev",
    -- snip --
    "process": {
      -- snip --
      "args": [
        "domainname"
      ],
      -- snip --
    },
    "hostname": "youki",
    "domainname": "youki_domain",
    "annotations": {},
    -- snip --
  }
  ```

- Execution result
  ```bash
   $ sudo ./youki run -b ./tutorial youki_test
   -- snip --
   youki_domain
  ```

<a href="https://gitpod.io/#https://github.com/containers/youki/pull/1214"></a>

### CODE REVIEW

**Comment by utam0k** [comment]
@higuruchi Can I ask you to put your signature into your commits?
```console
$ git commit --amend --no-edit -s
$ git push -f origin(or upstream?) HEAD
```

**Comment by utam0k** [comment]
What is the status of this PR? read for review? If not, please convert to draft PR.

File: crates/libcontainer/src/syscall/linux.rs [hunk 1]
```diff
@@ -198,6 +198,25 @@ impl Syscall for LinuxSyscall {
...
+        let res = unsafe { setdomainname(ptr, len) };
+        if res == EFAULT {
+            bail!("Failed to set {} as domainname. domainname pointed outside of user address space.", domainname)
+        }
+        if res == EINVAL {
+            bail!("Failed to set {} as domainname. domainname len was negative or too large.", domainname)
+        }
+        if res == EPERM {
+            bail!("Failed to set {} as domainname. the caller did not have the CAP_SYS_ADMIN capability.", domainname)
+        }
```
Comment (utam0k) [inline crates/libcontainer/src/syscall/linux.rs:215]: How about using `match` instead of if statements?

Comment (higuruchi) [reply crates/libcontainer/src/syscall/linux.rs]: Modified to match.
And when setdomainname syscall executed errno handling is invalid so fix it.

**Comment by YJDoc2** [comment]
Hey @higuruchi , can you fix the formatting issue that's failing the check? You will just need to run `cargo fmt` and push new commit / amend and force push

**Review by utam0k — APPROVED** [review]

**Comment by utam0k** [comment]
@higuruchi Thanks for your contribution!

**Pull request merged**